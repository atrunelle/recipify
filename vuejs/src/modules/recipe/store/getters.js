import * as types from './types';

export default {
  [types.GET_RECIPE]: (state) => state.recipeIngredients,
  [types.GET_RECIPES]: (state) => state.recipes,
};
