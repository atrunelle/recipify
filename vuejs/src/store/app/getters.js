import * as types from './types';

export default {
  [types.GET_ALERT]: (state) => state.alert,
};
