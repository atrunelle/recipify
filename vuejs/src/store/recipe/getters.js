import nutritionService from '@/core/nutrition.service';

export default {
  totalWeight: (state) => {
    return state.ingredients.reduce((sum, ingredient) => {
      return ingredient.nutrients.totalWeight + sum;
    }, 0);
  },

  totalCalories: (state) => {
    return state.ingredients.reduce((sum, ingredient) => {
      return ingredient.nutrients.calories + sum;
    }, 0);
  },

  totalNutrients: (state, getters) => {
    return nutritionService
      .getTotalNutrients(state.ingredients, getters.totalCalories);
  },
};
