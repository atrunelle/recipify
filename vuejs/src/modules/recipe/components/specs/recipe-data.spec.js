import Vuex from 'vuex';
import { mount, createLocalVue } from '@vue/test-utils';

import RecipeData from '../recipe-data';
import getters from '@/modules/recipe/store/getters';

describe('Component: Recipe data', () => {
  let component;
  let localVue = createLocalVue();

  localVue.use(Vuex);

  beforeEach(() => {
    const recipeStore = {
      namespaced: true,
      state: {
        recipeIngredients: [{
          nutrients: {
            totalNutrients: {
              FAT: {
                quantity: 5,
                unit: 'gr',
                label: 'Fat',
              },
              PROCNT: {
                quantity: 5,
                unit: 'gr',
                label: 'Fat',
              },
            },
            totalDaily: {
              FAT: {
                quantity: 5,
                unit: '%',
                label: 'Fat',
              },
              PROCNT: {
                quantity: 5,
                unit: '%',
                label: 'Fat',
              },
            },
          },
        }],
      },
      getters,
    };

    const store = new Vuex.Store({
      modules: {
        recipe: recipeStore,
      },
    });

    component = mount(RecipeData, {
      stubs: ['nutrients-chart'],
      localVue,
      store,
    });
  });

  it('should match snapshot', () => {
    expect(component.html()).toMatchSnapshot();
  });
});
