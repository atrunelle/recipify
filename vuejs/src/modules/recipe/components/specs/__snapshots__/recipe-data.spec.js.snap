// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Recipe data should match snapshot 1`] = `
<div>
    <div class="card" style="height: auto;">
        <div class="card__title card__title--primary">
            <h2>Total nutrients</h2>
        </div>
        <div class="card__text">
            <h3>Calories: NaNcal</h3>
            <h3>Total Weight: NaNgr</h3>
            <div class="layout align-end justify-space-between">
                <div class="flex text-xs-center">
                    <p>Total Nutrients</p>
                    <p>Total Daily percentage</p>
                </div>
                <div class="flex text-xs-center">
                    <h3 class="pa-2"></h3>
                    <p></p>
                    <p></p>
                </div>
                <div class="flex text-xs-center">
                    <h3 class="pa-2">Fat</h3>
                    <p>5.00gr</p>
                    <p>5.00%</p>
                </div>
                <div class="flex text-xs-center">
                    <h3 class="pa-2">Fat</h3>
                    <p>5.00gr</p>
                    <p>5.00%</p>
                </div>
            </div>
        </div>
    </div>
    <div class="card" style="height: auto;">
        <div class="card__title">
            <h2>Macro in recipe</h2>
        </div>
        <div class="card__text">
            <p>Based on 4 calories per gram of carbs and protein, and 9 calories per gram of fat.</p>
            <p>Those numbers are approximation only, as the number of calories per macro change per ingredient and depend on quality and other factor.</p>
        </div>
        <!---->
    </div>
</div>
`;
